### DISCUSSION TITLE
Stream support

### DISCUSSION THREAD

**Discussion by shiftrtech** [vercel/swr#709] [General]
Hi, i cant find examples of using SWR with a api endpoint with response type `stream`, i want to consuming the file parses progress, is this possible?, thanks.

**Comment by sergiodxa** [comment]
I think it will not be possible, SWR expects the data to be JSON and JSON (AFAIK) can't be easily streamed (because you need the closing part of an object or array). Also it expected the fetcher to resolve to a single value.

What you could do, is to have an endpoint to get the progress of that file and configure SWR to refetch it in an interval.
